tests/unit/test-cmd/test.run.js

```diff
@@ -282,6 +290,82 @@ describe('run', () => {
     });
   });
 
+  it('installs additional experiments addons', async () => {
+    const experiments = [
+      fixturePath('fake-experiment'),
+      fixturePath('fake-experiment2'),
+    ];
+    const experimentsResults = [
+      {addon: {id: 'expnamespace@experiments.addons.mozilla.org'}},
+      {addon: {id: 'expnamespace2@experiments.addons.mozilla.org'}},
+    ];
+
+    const cmd = prepareRun(null, experiments);
+
+    let installCallIndex = 0;
+    const firefoxClient = fake(RemoteFirefox.prototype, {
+      installTemporaryAddon: () => {
+        if (installCallIndex < experimentsResults.length) {
+          return Promise.resolve(experimentsResults[installCallIndex++]);
+        }
+
+        return Promise.resolve(tempInstallResult);
+      },
+    });
+
+    await cmd.run({}, {
+      firefoxClient: sinon.spy(() => Promise.resolve(firefoxClient)),
+    });
+
+    assert.equal(firefoxClient.installTemporaryAddon.callCount, 3);
+    assert.equal(firefoxClient.installTemporaryAddon.firstCall.args[0],
+                 path.resolve(experiments[0]));
+    assert.equal(firefoxClient.installTemporaryAddon.secondCall.args[0],
+                 path.resolve(experiments[1]));
+    assert.equal(firefoxClient.installTemporaryAddon.lastCall.args[0],
+                 cmd.argv.sourceDir);
+  });
+
+  it('raise an error on missing experiment addonId', async () => {
+    const experiments = [
+      fixturePath('fake-experiment'),
+      fixturePath('fake-experiment2'),
+    ];
+    const experimentsResults = [
+      {addon: {}},
+      {addon: {id: 'expnamespace2@experiments.addons.mozilla.org'}},
+    ];
+
+    const cmd = prepareRun(null, experiments);
+
+    let installCallIndex = 0;
+    const firefoxClient = fake(RemoteFirefox.prototype, {
+      installTemporaryAddon: () => {
+        if (installCallIndex < experimentsResults.length) {
+          return Promise.resolve(experimentsResults[installCallIndex++]);
+        }
+
+        return Promise.resolve(tempInstallResult);
+      },
+    });
+
+    assert.isRejected(cmd.run({}, {
+      firefoxClient: sinon.spy(() => Promise.resolve(firefoxClient)),
+    }), `Missing ${experiments[0]} addonId on installExperiments`);
+  });
+
+  it('raises an error message on --experiments and --pre-install', () => {
+    const experiments = [
+      fixturePath('fake-experiment'),
+      fixturePath('fake-experiment2'),
+    ];
+
+    const cmd = prepareRun(null, experiments);
+
+    assert.isRejected(cmd.run({preInstall: true}), UsageError);
```

Comment:
Since there are many possible UsageErrors I think it would help to add a regex to match a bit of the error message